spec/finance/api_client_spec.rb

```diff
@@ -0,0 +1,187 @@
+
+      expect(api_client.json_mime?('application/json')).to eq true
+      expect(api_client.json_mime?('application/json; charset=UTF8')).to eq true
+      expect(api_client.json_mime?('APPLICATION/JSON')).to eq true
+
+      expect(api_client.json_mime?('application/xml')).to eq false
+      expect(api_client.json_mime?('text/plain')).to eq false
+      expect(api_client.json_mime?('application/jsonp')).to eq false
+    end
+  end
+
+  describe '#select_header_accept' do
+    let(:api_client) { XeroRuby::Finance::ApiClient.new }
+
+    it 'works' do
+      expect(api_client.select_header_accept(nil)).to be_nil
+      expect(api_client.select_header_accept([])).to be_nil
+
+      expect(api_client.select_header_accept(['application/json'])).to eq('application/json')
+      expect(api_client.select_header_accept(['application/xml', 'application/json; charset=UTF8'])).to eq('application/json; charset=UTF8')
+      expect(api_client.select_header_accept(['APPLICATION/JSON', 'text/html'])).to eq('APPLICATION/JSON')
+
+      expect(api_client.select_header_accept(['application/xml'])).to eq('application/xml')
+      expect(api_client.select_header_accept(['text/html', 'application/xml'])).to eq('text/html,application/xml')
+    end
+  end
+
+  describe '#select_header_content_type' do
+    let(:api_client) { XeroRuby::Finance::ApiClient.new }
+
+    it 'works' do
+      expect(api_client.select_header_content_type(nil)).to eq('application/json')
+      expect(api_client.select_header_content_type([])).to eq('application/json')
+
+      expect(api_client.select_header_content_type(['application/json'])).to eq('application/json')
+      expect(api_client.select_header_content_type(['application/xml', 'application/json; charset=UTF8'])).to eq('application/json; charset=UTF8')
+      expect(api_client.select_header_content_type(['APPLICATION/JSON', 'text/html'])).to eq('APPLICATION/JSON')
+      expect(api_client.select_header_content_type(['application/xml'])).to eq('application/xml')
+      expect(api_client.select_header_content_type(['text/plain', 'application/xml'])).to eq('text/plain')
+    end
+  end
+
+  describe '#sanitize_filename' do
+    let(:api_client) { XeroRuby::Finance::ApiClient.new }
+
+    it 'works' do
+      expect(api_client.sanitize_filename('sun')).to eq('sun')
+      expect(api_client.sanitize_filename('sun.gif')).to eq('sun.gif')
+      expect(api_client.sanitize_filename('../sun.gif')).to eq('sun.gif')
+      expect(api_client.sanitize_filename('/var/tmp/sun.gif')).to eq('sun.gif')
+      expect(api_client.sanitize_filename('./sun.gif')).to eq('sun.gif')
+      expect(api_client.sanitize_filename('..\sun.gif')).to eq('sun.gif')
+      expect(api_client.sanitize_filename('\var\tmp\sun.gif')).to eq('sun.gif')
+      expect(api_client.sanitize_filename('c:\var\tmp\sun.gif')).to eq('sun.gif')
+      expect(api_client.sanitize_filename('.\sun.gif')).to eq('sun.gif')
+    end
+  end
+end
```

Comment:
I don't think this file should exist. Looking at the other API sets they only have API spec files no client spec file

Reply:
FYI - Yeah the spec files for a whole new API set need to be trimmed up a bit once they are copied over from the scripts only the initial time. If directions are not clear in the confluence happy to consult.

In addition, based on how the ruby class/constant namespace works, some constants are duplicated across the OAS such as organisation_type and entity_type both have ACCOUNTING_PRACTICE

https://github.com/XeroAPI/xero-ruby/blob/master/lib/xero-ruby/models/accounting/organisation.rb#L137

![Screen Shot 2021-11-10 at 2 15 31 PM](https://user-images.githubusercontent.com/2935387/141194624-35b69019-d9e2-4bdb-9fa3-d75f0f76fca6.png)
.

Reply:
Thanks Chris, did not realise that.  Appreciate the pointer!

I've handled the constants in an upstream fix in the generators, pending merge.